### PR TITLE
qa/tasks/vstart_runner,doc: be compatible with python3

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/dev/developer_guide/running-tests-locally.rst [hunk 1]
```diff
@@ -50,17 +52,12 @@ Running your first test
 The Python tests in Ceph repository can be executed on your local machine
 using `vstart_runner.py`_. To do that, you'd need `teuthology`_ installed::
 
-    $ git clone https://github.com/ceph/teuthology
-    $ cd teuthology/
-    $ virtualenv -p python2.7 ./venv
+    $ virtualenv --python=python3 venv
```
Comment (alfonsomthd) [inline doc/dev/developer_guide/running-tests-locally.rst:55]: @tchaikov For consistency, can you also update this line
https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/run-backend-api-tests.sh#L44
to sth like:
```
virtualenv --python=${TEUTHOLOGY_PYTHON_BIN:-/usr/bin/python3} venv
```

Comment (tchaikov) [reply doc/dev/developer_guide/running-tests-locally.rst:55]: sure. updated and repushed.

**Comment by tchaikov** [comment]
jenkins render docs

**Comment by lxbsz** [comment]
@tchaikov 

Run `# python3 ../qa/tasks/vstart_runner.py tasks.cephfs.test_volume_client.TestVolumeClient.test_get_authorized_ids` failed for me.

I have paste the fixing patch, could you fix it here ?

```
diff --git a/qa/tasks/cephfs/fuse_mount.py b/qa/tasks/cephfs/fuse_mount.py
index 227d86e..cd05824 100644
--- a/qa/tasks/cephfs/fuse_mount.py
+++ b/qa/tasks/cephfs/fuse_mount.py
@@ -454,9 +454,9 @@ print(find_socket("{client_name}"))
 
         # Query client ID from admin socket
         json_data = self.client_remote.sh(
-            ['sudo', self._prefix + 'ceph', '--admin-daemon', asok_path] + args,
+            ['sudo', self._prefix + 'ceph', '--admin-daemon', asok_path.decode("utf-8")] + args,
             timeout=(15*60))
-        return json.loads(json_data)
+        return json.loads(json_data.decode("utf-8"))
 
     def get_global_id(self):
         """
diff --git a/qa/tasks/vstart_runner.py b/qa/tasks/vstart_runner.py
index 5dec6a6..c73ba04 100644
--- a/qa/tasks/vstart_runner.py
+++ b/qa/tasks/vstart_runner.py
@@ -390,7 +390,7 @@ class LocalRemote(object):
 
             # Hack: writing to stdin is not deadlock-safe, but it "always" works
             # as long as the input buffer is "small"
-            subproc.stdin.write(stdin)
+            subproc.stdin.write(stdin.encode())
 
         proc = LocalRemoteProcess(
             args, subproc, check_status,
```

**Comment by tchaikov** [comment]
@lxbsz instead i am passing `StringIO()` for capturing stdout.

**Pull request merged**